app/lib/registries/world_locations_registry.rb

```diff
@@ -0,0 +1,37 @@
+module Registries
+  class WorldLocationsRegistry
+    CACHE_KEY = "registries/world_locations".freeze
+
+    def all
+      @all ||= cached_locations
+    end
+
+    def [](slug)
+      all.find { |o| o['slug'] == slug }
+    end
+
+  private
+
+    def cached_locations
+      Rails.cache.fetch(CACHE_KEY, expires_in: 1.hour) do
+        fetch_locations
+      end
+    rescue GdsApi::HTTPServerError, GdsApi::HTTPBadGateway
+      GovukStatsd.increment "ERROR: Could not get world locations from the API!"
```

Comment:
I think the text should be like a key and you can namespace it if you want.  Something like `registries.world_location_api_errors` would work? Here's an example of an existing one: https://github.com/alphagov/signon/blob/d8b0202e1c601045560845c42d86c296c4adb5bd/app/controllers/passwords_controller.rb#L41 (sorry 🙈 ).

You can navigate the ones we have at the moment under `stats.govuk.app....` on https://graphite.publishing.service.gov.uk/dashboard/

Reply:
Aha, thank you very much 😃